Failed tests (10669): TestReplicateRogueRemovedNode

The following test appears to have failed:

[#10669](https://circleci.com/gh/cockroachdb/cockroach/10669):

```
I0106 11:16:27.819306 937 kv/range_cache.go:189  adding descriptor: key=/Meta2/Max desc=range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I0106 11:16:27.821219 937 storage/raft_log_queue.go:133  truncating the raft log of range 1 to 16
I0106 11:16:27.825337 937 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
I0106 11:16:27.825565 937 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
I0106 11:16:27.825719 937 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- FAIL: TestRaftLogQueue (0.03s)
	client_raft_log_queue_test.go:107: raft log was truncated again and it shouldn't have been, afterTruncationIndex:14 after2ndTruncationIndex:16
=== RUN   TestStoreRecoverFromEngine
I0106 11:16:27.827265 937 gossip/gossip.go:193  setting node descriptor node_id:1 address:<network_field:"localhost" address_field:"1" > attrs:<> 
I0106 11:16:27.830056 937 raft/raft.go:446  [group 1] 1 became follower at term 5
I0106 11:16:27.830252 937 raft/raft.go:234  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I0106 11:16:27.830350 937 raft/raft.go:526  [group 1] 1 is starting a new election at term 5
I0106 11:16:27.830451 937 raft/raft.go:459  [group 1] 1 became candidate at term 6
I0106 11:16:27.830558 937 raft/raft.go:508  [group 1] 1 received vote from 1 at term 6
I0106 11:16:27.830675 937 raft/raft.go:487  [group 1] 1 became leader at term 6
I0106 11:16:27.833305 937 storage/replica_command.go:1199  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1.000s
--
I0106 11:19:40.775329 937 storage/replica.go:1558  gossiping cluster id  from store 2, range 1
I0106 11:20:40.770658 937 storage/replica.go:1558  gossiping cluster id  from store 1, range 1
I0106 11:20:40.771008 937 storage/replica.go:1567  gossiping sentinel from store 1, range 1
I0106 11:20:40.771272 937 storage/replica.go:1573  gossiping first range from store 1, range 1
I0106 11:20:40.782941 937 storage/replica.go:1558  gossiping cluster id  from store 2, range 1
panic: test timed out after 5m0s

goroutine 9219 [running]:
testing.startAlarm.func1()
	/tmp/workdir/go/src/testing/testing.go:703 +0x16b
created by time.goFunc
	/tmp/workdir/go/src/time/sleep.go:129 +0x6e

goroutine 1 [chan receive, 4 minutes]:
testing.RunTests(0x1a69570, 0x21162a0, 0xdb, 0xdb, 0x0)
	/tmp/workdir/go/src/testing/testing.go:562 +0xafa
testing.(*M).Run(0xc820032f10, 0x1)
	/tmp/workdir/go/src/testing/testing.go:494 +0xe5
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc820032f10)
	/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:34 +0x2f
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc820032f10)
	/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:36 +0x34
main.main()
	github.com/cockroachdb/cockroach/storage/_test/_testmain.go:498 +0x20a

goroutine 17 [syscall, 4 minutes, locked to thread]:
runtime.goexit()
	/tmp/workdir/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 19 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x21714e0)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1029 +0x76
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0x117

goroutine 9050 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*Replica).addWriteCmd(0xc8204dadd0, 0x7f9c95ffa6f0, 0xc8204c1aa0, 0x0, 0x88, 0x0, 0x0, 0x1, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:926 +0x7be
github.com/cockroachdb/cockroach/storage.(*Replica).Send(0xc8204dadd0, 0x7f9c95ffa6f0, 0xc8204c1aa0, 0x0, 0x88, 0x0, 0x0, 0x1, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:628 +0x6a3
github.com/cockroachdb/cockroach/storage.(*Store).Send(0xc8200b8000, 0x7f9c95ffa6f0, 0xc8204c1aa0, 0x0, 0x88, 0x0, 0x0, 0x1, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1312 +0xd81
github.com/cockroachdb/cockroach/client.Wrap.func1(0x7f9c95ffa6b0, 0xc82007d6c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/client/sender.go:109 +0xc5
github.com/cockroachdb/cockroach/client.SenderFunc.Send(0xc8204799e0, 0x7f9c95ffa6b0, 0xc82007d6c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/client/sender.go:44 +0x59
--
testing.tRunner(0xc82008e2d0, 0x21174d0)
	/tmp/workdir/go/src/testing/testing.go:456 +0xdd
created by testing.RunTests
	/tmp/workdir/go/src/testing/testing.go:561 +0xaa4

goroutine 9056 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1614 +0xb0c
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201eea10, 0xc8204b14b0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9099 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc8201b4aa0)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x5b
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x581

goroutine 9052 [select]:
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).startStats(0xc8202de640)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:214 +0xf1a
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).(github.com/cockroachdb/cockroach/kv.startStats)-fm()
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:201 +0x2e
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ee460, 0xc8204b0270)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9082 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:319 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201eea10, 0xc820422a00)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9107 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:319 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82016e9a0, 0xc8204990e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9094 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:556 +0x15d
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82016e9a0, 0xc820498b80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9090 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc8201b4780)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x5b
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x581

goroutine 9089 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/rpc.(*Server).sendResponses(0xc82005dbc0, 0x7f9c97061340, 0xc820376180, 0xc820404d20)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:300 +0xb5
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func1(0xc82005dbc0, 0x7f9c97061340, 0xc820376180, 0xc820404d20)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:207 +0x4d
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:208 +0xa6c

goroutine 9079 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:319 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201eea10, 0xc8204229a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9138 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f9c95ff9b08, 0x72, 0x0)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc8201efe90, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc8201efe90, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).Read(0xc8201efe30, 0xc8203a6000, 0x1000, 0x1000, 0x0, 0x7f9c95fed028, 0xc82007c0b0)
	/tmp/workdir/go/src/net/fd_unix.go:232 +0x27b
net.(*conn).Read(0xc8200262d0, 0xc8203a6000, 0x1000, 0x1000, 0xc820315900, 0x0, 0x0)
	/tmp/workdir/go/src/net/net.go:172 +0x121
--
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func2(0xc82005dbc0, 0x7f9c960036e0, 0xc8200262d0, 0x7f9c97061340, 0xc820376180, 0xc82048c050, 0xc820404d20)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:210 +0x6f
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:213 +0xadc

goroutine 9057 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:556 +0x15d
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201eea10, 0xc820422800)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9101 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/util.ListenAndServe.func2()
	/go/src/github.com/cockroachdb/cockroach/util/net.go:88 +0x7b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ee460, 0xc820498f80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9124 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f9c95ff9bc8, 0x72, 0x0)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc82016fcd0, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc82016fcd0, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).Read(0xc82016fc70, 0xc82051b000, 0x1000, 0x1000, 0x0, 0x7f9c95fed028, 0xc82007c0b0)
	/tmp/workdir/go/src/net/fd_unix.go:232 +0x27b
net.(*conn).Read(0xc820080400, 0xc82051b000, 0x1000, 0x1000, 0x7f9c970614f8, 0x0, 0x0)
	/tmp/workdir/go/src/net/net.go:172 +0x121
--
net/rpc.(*Client).input(0xc8204c8e40)
	/usr/local/go/src/net/rpc/client.go:109 +0x177
created by net/rpc.NewClientWithCodec
	/usr/local/go/src/net/rpc/client.go:201 +0x126

goroutine 9111 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:319 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82016e9a0, 0xc820499180)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9091 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f9c95ffa1c8, 0x72, 0x7f9c95fed028)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc82016eb50, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc82016eb50, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).accept(0xc82016eaf0, 0x0, 0x7f9c97053520, 0xc820422b20)
	/tmp/workdir/go/src/net/fd_unix.go:408 +0x2f6
net.(*TCPListener).AcceptTCP(0xc820080358, 0xc82030ad20, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:254 +0x77
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ee460, 0xc82047ce40)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9054 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f9c95ffa048, 0x72, 0x7f9c95fed028)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc8201ef4f0, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc8201ef4f0, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).accept(0xc8201ef490, 0x0, 0x7f9c97053520, 0xc820422e40)
	/tmp/workdir/go/src/net/fd_unix.go:408 +0x2f6
net.(*TCPListener).AcceptTCP(0xc820026298, 0xc820307d20, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:254 +0x77
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ee460, 0xc820475890)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9081 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:319 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201eea10, 0xc8204229e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9092 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/util.ListenAndServe.func2()
	/go/src/github.com/cockroachdb/cockroach/util/net.go:88 +0x7b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ee460, 0xc820498b40)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9100 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f9c95ff9d48, 0x72, 0x7f9c95fed028)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc82016eed0, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc82016eed0, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).accept(0xc82016ee70, 0x0, 0x7f9c97053520, 0xc8204d0480)
	/tmp/workdir/go/src/net/fd_unix.go:408 +0x2f6
net.(*TCPListener).AcceptTCP(0xc820080378, 0xc820037d20, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:254 +0x77
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ee460, 0xc82047d590)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9053 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc8204a2190)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x5b
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x581

goroutine 9113 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc8204dca80, 0xece1eee98, 0x2ea9b671, 0x2170f60, 0xc820474e70, 0xc82016e9a0, 0x0, 0x50)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:176 +0x641
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:213 +0x1b0
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82016e9a0, 0xc8204991e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9095 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:621 +0x447
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82016e9a0, 0xc820498ba0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9139 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f9c95ffa588, 0x72, 0x0)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc8201efe20, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc8201efe20, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).Read(0xc8201efdc0, 0xc82049e000, 0x1000, 0x1000, 0x0, 0x7f9c95fed028, 0xc82007c0b0)
	/tmp/workdir/go/src/net/fd_unix.go:232 +0x27b
net.(*conn).Read(0xc8200262c8, 0xc82049e000, 0x1000, 0x1000, 0x7f9c970614f8, 0x0, 0x0)
	/tmp/workdir/go/src/net/net.go:172 +0x121
--
net/rpc.(*Client).input(0xc820404de0)
	/usr/local/go/src/net/rpc/client.go:109 +0x177
created by net/rpc.NewClientWithCodec
	/usr/local/go/src/net/rpc/client.go:201 +0x126

goroutine 9055 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/util.ListenAndServe.func2()
	/go/src/github.com/cockroachdb/cockroach/util/net.go:88 +0x7b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ee460, 0xc820422780)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9085 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/rpc.(*Server).sendResponses(0xc8204dcd80, 0x7f9c97061340, 0xc820376000, 0xc820404540)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:300 +0xb5
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func1(0xc8204dcd80, 0x7f9c97061340, 0xc820376000, 0xc820404540)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:207 +0x4d
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:208 +0xa6c

goroutine 9077 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:319 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201eea10, 0xc820422960)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9078 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:319 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201eea10, 0xc820422980)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9096 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:640 +0x447
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82016e9a0, 0xc820498bc0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9112 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:319 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82016e9a0, 0xc8204991c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9051 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*StorePool).start.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store_pool.go:213 +0x208
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ee460, 0xc8204221a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9080 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:319 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201eea10, 0xc8204229c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9110 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:319 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82016e9a0, 0xc820499160)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9093 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1614 +0xb0c
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82016e9a0, 0xc82049b530)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9108 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:319 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82016e9a0, 0xc820499120)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9075 [select, 2 minutes]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:640 +0x447
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201eea10, 0xc8204228c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9083 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc82005d380, 0xece1eee98, 0x2eea1591, 0x2170f60, 0xc820474e70, 0xc8201eea10, 0xc8204dadd0, 0xc8200b81c0)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:176 +0x641
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1.1(0xc8204dadd0, 0xc8204747e0)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:217 +0xb0
github.com/cockroachdb/cockroach/storage.(*storeRangeSet).Visit(0xc820474870, 0xc8204759b0)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:225 +0x45c
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:219 +0x452
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201eea10, 0xc820422a20)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9109 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:319 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82016e9a0, 0xc820499140)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9074 [select, 2 minutes]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:621 +0x447
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201eea10, 0xc8204228a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 9086 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f9c95ffa348, 0x72, 0x0)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc8201ef560, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc8201ef560, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).Read(0xc8201ef500, 0xc820019000, 0x1000, 0x1000, 0x0, 0x7f9c95fed028, 0xc82007c0b0)
	/tmp/workdir/go/src/net/fd_unix.go:232 +0x27b
net.(*conn).Read(0xc8200262a8, 0xc820019000, 0x1000, 0x1000, 0xc820313900, 0x0, 0x0)
	/tmp/workdir/go/src/net/net.go:172 +0x121
--
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:233 +0x130
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func2(0xc8204dcd80, 0x7f9c960036e0, 0xc8200262a8, 0x7f9c97061340, 0xc820376000, 0xc8204b1610, 0xc820404540)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:210 +0x6f
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:213 +0xadc
FAIL	github.com/cockroachdb/cockroach/storage	300.084s
=== RUN   TestBatchBasics
I0106 11:16:40.522936 995 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.01s)
=== RUN   TestBatchGet
I0106 11:16:40.527124 995 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN   TestBatchMerge
I0106 11:16:40.534864 995 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.00s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

tl;dr `TestReplicateRogueRemovedNode`

```
I0107 01:49:19.567952 948 storage/replica.go:996  raft: proposing REMOVE_REPLICA {3 3 3} for range 1
W0107 01:49:19.572371 948 storage/replica.go:1156  group 1 on store 1 failed to send message to 3: storage/raft_transport.go:163: unknown peer 3
I0107 01:49:19.576646 948 storage/replica_command.go:450  auto-gc'ed 142512f7 (3 intents)
W0107 01:49:19.578979 948 storage/replica.go:1156  group 1 on store 1 failed to send message to 3: storage/raft_transport.go:163: unknown peer 3
I0107 01:49:19.580990 948 storage/replica_command.go:450  auto-gc'ed 142512f7 (3 intents)
I0107 01:49:19.585077 948 storage/replica.go:996  raft: proposing REMOVE_REPLICA {2 2 2} for range 1
I0107 01:49:19.591711 948 storage/replica_command.go:450  auto-gc'ed 0e6db142 (3 intents)
I0107 01:49:19.593377 948 storage/replica_gc_queue.go:168  unable to acquire lease from valid range range=1 [/Min-/Max): range 1 was not found
I0107 01:49:19.597951 948 storage/replica_command.go:450  auto-gc'ed 0e6db142 (3 intents)
E0107 01:49:19.600431 948 storage/store.go:1622  error handling raft message: raft: cannot step as peer not found
E0107 01:49:19.603054 948 storage/store.go:1622  error handling raft message: raft: cannot step as peer not found
I0107 01:50:19.415623 948 storage/replica.go:1558  gossiping cluster id  from store 2, range 1
I0107 01:50:19.417942 948 storage/replica.go:1558  gossiping cluster id  from store 1, range 1
I0107 01:50:19.418267 948 storage/replica.go:1567  gossiping sentinel from store 1, range 1
I0107 01:50:19.418519 948 storage/replica.go:1573  gossiping first range from store 1, range 1
I0107 01:51:19.415557 948 storage/replica.go:1558  gossiping cluster id  from store 2, range 1
I0107 01:51:19.420642 948 storage/replica.go:1558  gossiping cluster id  from store 1, range 1
I0107 01:51:19.420953 948 storage/replica.go:1567  gossiping sentinel from store 1, range 1
I0107 01:51:19.421255 948 storage/replica.go:1573  gossiping first range from store 1, range 1
I0107 01:52:19.415716 948 storage/replica.go:1558  gossiping cluster id  from store 2, range 1
I0107 01:52:19.418948 948 storage/replica.go:1558  gossiping cluster id  from store 1, range 1
I0107 01:52:19.419366 948 storage/replica.go:1567  gossiping sentinel from store 1, range 1
I0107 01:52:19.419940 948 storage/replica.go:1573  gossiping first range from store 1, range 1
I0107 01:53:19.415419 948 storage/replica.go:1558  gossiping cluster id  from store 2, range 1
I0107 01:53:19.418029 948 storage/replica.go:1558  gossiping cluster id  from store 1, range 1
I0107 01:53:19.418330 948 storage/replica.go:1567  gossiping sentinel from store 1, range 1
I0107 01:53:19.418552 948 storage/replica.go:1573  gossiping first range from store 1, range 1
panic: test timed out after 5m0s
```

--- a/GLOCKFILE
+++ b/GLOCKFILE
@@ -21,7 +21,7 @@ github.com/cockroachdb/c-rocksdb b7fb7bddcb55be35eacdf67e9e2c931083ce02c4
 github.com/cockroachdb/c-snappy 5c6d0932e0adaffce4bfca7bdf2ac37f79952ccf
 github.com/cockroachdb/yacc 443154b1852a8702b07d675da6cd97cd9177a316
 github.com/codahale/hdrhistogram 954f16e8b9ef0e5d5189456aa4c1202758e04f17
-github.com/coreos/etcd 22dd738228b60c5d55ce7a0f8fc0499ae9911910
+github.com/coreos/etcd 8199147cf859882f625523446c28ae2e53b2432f
 github.com/docker/go-units 0bbddae09c5a5419a8c6dcdd7ff90da3d450393b
 github.com/elazarl/go-bindata-assetfs 57eb5e1fc594ad4b0b1dbea7b286d299e0cb43c2
 github.com/go-sql-driver/mysql 6fd058ce0d6b7ee43174e80d5a3e7f483c4dfbe5

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -1255,7 +1255,6 @@ func TestStoreRangeRebalance(t *testing.T) {
 // cannot cause other removed nodes to recreate their ranges.
 func TestReplicateRogueRemovedNode(t *testing.T) {
 	defer leaktest.AfterTest(t)
-	t.Skip("#3642")
 
 	mtc := startMultiTestContext(t, 3)
 	defer mtc.Stop()

